### PR TITLE
init ar1_whitenoise

### PR DESCRIPTION
**Pull request by JanJereczek**
Implemented:
- ar1_whitenoise function to estimate the AR1 regression coefficient based on alanytical solution of least-square
- test with a known AR1 process
- benchmark to test 0 allocation

### CODE REVIEW

**Comment by Datseris** [comment]
@JanJereczek maybe you should add in this PR a function that actually uses the output of AR1 to detect a transition or something. Like detect significant changes in AR1 or something like that?

File: test/indicators.jl [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+function g_whitenoise!(dx, x, p, t)
+    dx[1] = p[2]
+end
+
+function generate_ar1(x0::Vector{T}, p::Vector{T}, t::Vector{T}) where {T<:Real}
+    tspan = extrema(t)
+    prob = SDEProblem(f_ar1!, g_whitenoise!, x0, tspan, p)
+    sol = solve(prob, EM(), dt=t[2]-t[1])
+    return hcat(sol.(t)...)
+end
```
Comment (Datseris) [inline test/indicators.jl:20]: Is this the most efficient way to generate AR1? We are talking about autoregressive noise of order 1 here, right? Or does AR1 mean something else to you? Here is how we do it in TimeseriesSurrogates.jl: https://github.com/JuliaDynamics/TimeseriesSurrogates.jl/blob/main/src/utils/testsystems.jl#L14-L23

**Comment by Datseris** [comment]
We should have one `change_point_detection` function that searches for significant differences in the observable over a rolling window. Here the observable is just AR1, but it could also be permutation entropy. Later, we can add a different function `early_warning_signal` that detects smooth increase of the observable. but for now lets keep it simple with the `change_point_function`?

File: test/indicators.jl (hunk 1 above)
Comment (JanJereczek) [reply test/indicators.jl]: `AR1()` from TimeseriesSurrogates.jl is now used in the test instead of its DifferentialEquations.jl equivalent :)

**Pull request merged**